### ISSUE TITLE
Godot 4.1 will have breaking changes to gdextension

### ISSUE THREAD

**Issue by nvanfleet** [migueldeicaza/SwiftGodot#95]
Just a note that it seems like there will be breaking changes coming in Godot 4.1 to GDExtension
https://godotengine.org/article/dev-snapshot-godot-4-1-beta-2/

```
GDExtension: Standardize Object ptrcall encoding on Object ** ([GH-77410](https://github.com/godotengine/godot/pull/77410)).
This change was necessary to fix a number of crashes, but it breaks compatibility with Godot 4.0’s GDExtension API in a way that we cannot easily work around. This was evaluated as a necessary tradeoff to make GDExtension usable in production, but it means that you will need to keep separate branches or add your own compatibility code for Godot 4.0 and Godot 4.1+ GDExtension interfaces if you want to support both.
```

**Comment by hyouuu** [comment]
About to dive into game development and this Swift plugin has been our most wanted thing with Godot - @migueldeicaza thanks for making this! It seems you haven't been active on this for a month, and wondering is there any plans and rough timeline when it would support 4.1?

**Comment by nvanfleet** [comment]
> About to dive into game development and this Swift plugin has been our most wanted thing with Godot - @migueldeicaza thanks for making this! It seems you haven't been active on this for a month, and wondering is there any plans and rough timeline when it would support 4.1?

He is away for a bit but will be back. I just wanted to leave this note here. I haven't checked 4.1 yet.

**Comment by migueldeicaza** [comment]
Yeah on a long vacation with the family.   It shouldn’t be too hard to fix - specially now that I realized how much faster an arm Mac is vs my old intel

**Comment by hyouuu** [comment]
Congrats on the new Mac!

**Comment by migueldeicaza** [comment]
I could not figure out where the `ptrcall` affects the binding yet, but I did the rest of the work necessary to support 4.1, and could use someone to help me find where this is crashing (so I can track down this `ptrcall` issue):

https://github.com/migueldeicaza/SwiftGodot/pull/104

**Comment by migueldeicaza** [comment]
Addressed in the 4.1 branch
